modules/php/States/RockPaperScissorsTrait.php

```diff
@@ -0,0 +1,193 @@
+  define("RPS_OPTION_SCISSORS", 3);
+}
+
+trait RockPaperScissorsTrait
+{
+  private function getOptionLabels()
+  {
+    $options = [];
+
+    $options[RPS_OPTION_ROCK] = clienttranslate("Rock");
+    $options[RPS_OPTION_PAPER] = clienttranslate("Paper");
+    $options[RPS_OPTION_SCISSORS] = clienttranslate("Scissors");
+
+    return $options;
+  }
+
+  private function checkChallengerWins($challenger, $challenged)
+  {
+    // Rock beats Scissors
+    if ($challenger == RPS_OPTION_ROCK && $challenged == RPS_OPTION_SCISSORS) {
+      return true;
+    }
+    // Scissors beats Paper
+    if ($challenger == RPS_OPTION_SCISSORS && $challenged == RPS_OPTION_PAPER) {
+      return true;
+    }
+    // Paper beats Rock
+    if ($challenger == RPS_OPTION_PAPER && $challenged == RPS_OPTION_ROCK) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public function st_nextRoundRockPaperScissors()
+  {
+    $options = $this->getOptionLabels();
+    $challenger_choice = self::getGameStateValue("rpsChallengerChoice");
+    $challenged_choice = self::getGameStateValue("rpsChallengedChoice");
+
+    // need distinct player choices, otherwies tie
+    if ($challenger_choice == $challenged_choice) {
+      self::notifyAllPlayers(
+        "rockPaperScissorsRound",
+        clienttranslate(
+          'Tie: ${challenger_choice} challenged ${challenged_choice}, try again'
+        ),
+        [
+          "challenger_choice" => $options[$challenger_choice],
+          "challenged_choice" => $options[$challenged_choice],
+        ]
+      );
+
+      $this->gamestate->nextstate("rockPaperScissorsContinue");
+      return;
+    }
+
+    $challenger_player_id = self::getGameStateValue("rpsChallengerId");
+    $challenged_player_id = self::getGameStateValue("rpsChallengedId");
+
+    $maxWins = 0;
+    $winning_player_id = -1;
+    $losing_player_id = -1;
+    // determine the winner and loser
+    $challengerWins = $this->checkChallengerWins(
+      $challenger_choice,
+      $challenged_choice
+    );
+    if ($challengerWins) {
+      $maxWins = self::incGameStateValue("rpsChallengerWins", 1);
+      $winning_player_id = $challenger_player_id;
+      $losing_player_id = $challenged_player_id;
+    } else {
+      $maxWins = self::incGameStateValue("rpsChallengedWins", 1);
+      $winning_player_id = $challenged_player_id;
+      $losing_player_id = $challenger_player_id;
+    }
+
+    $players = self::loadPlayersBasicInfos();
+    self::notifyAllPlayers(
+      "rockPaperScissorsRound",
+      clienttranslate(
+        '${player_name} wins: ${challenger_choice} challenged ${challenged_choice}'
+      ),
+      [
+        "player_id" => $winning_player_id,
+        "player_name" => $players[$winning_player_id]["player_name"],
+        "challenger_choice" => $options[$challenger_choice],
+        "challenged_choice" => $options[$challenged_choice],
+      ]
+    );
+
+    // as long as neither has won the best of 3, keep playing (next round)
```

Comment:
This could become best of 4 win with inflation. So you could win 2, then lose 2 and would need to do a 5th round.

Reply:
Oh wow, I didn't think about inflation here at all.
But it is a number 3 on the card, so I guess you are right - even though a "4-round Rock-Paper-Scissors tournament" doesn't make any sense so would indeed need to go up to 5.

Reply:
Fixed now